File: Medium/2026-04-22-2452-Words-Within-Two-Edits-of-Dictionary/solution.js
var twoEditWords = function(queries, dictionary) {
    const isWithinTwoEdits = (word1, word2) => {
        let diffCount = 0;
        for (let i = 0; i < word1.length; i++) {
            if (word1[i] !== word2[i]) {
                diffCount++;
                if (diffCount > 2) return false;
            }
        }
        return diffCount <= 2;
    };

    const result = [];
    for (const query of queries) {
        for (const dictWord of dictionary) {
            if (isWithinTwoEdits(query, dictWord)) {
                result.push(query);
                break;
            }
        }
    }
    return result;
};

File: Medium/2026-04-22-2452-Words-Within-Two-Edits-of-Dictionary/solution.py
class Solution:
    def twoEditWords(self, queries: List[str], dictionary: List[str]) -> List[str]:
        def is_within_two_edits(query: str, word: str) -> bool:
            diff_count = sum(1 for a, b in zip(query, word) if a != b)
            return diff_count <= 2
        
        result = []
        for query in queries:
            if any(is_within_two_edits(query, word) for word in dictionary):
                result.append(query)
        return result